astropy/utils/data.py

```diff
@@ -502,7 +502,7 @@ def get_pkg_data_fileobj(data_name, package=None, encoding=None, cache=True):
     get_pkg_data_filename : returns a local name for a file containing the data
     """  # noqa
 
-    datafn = _find_pkg_data_path(data_name, package=package)
+    datafn = get_pkg_data_path(data_name, package=package)
```

Comment:
Are you very sure this won't accidentally introduce unwanted API change?

Reply:
The only different between the two functions is that ``get_pkg_data_path`` performs a string join on the path. Since `_find_pkg_data_path` only accepts 1 argument for the path, there are two places in the code to check that there should be no API change:

1)
```python
if module is None:
    # return data_name  # old
    return os.path.join(*path)  # path is always a tuple, b/c it's an *args
```

```python
>>> data_name = "some/path"
>>> path = (data_name, )  # mimicking passing through func(*args)
>>> os.path.join(*path)
'some/path'
```

2)
```python
# full_path = os.path.join(module_path, data_name)   # old
full_path = os.path.join(module_path, *path)
```

```python
>>> module_path = "~/astropy"
>>> data_name = "some/path"
>>> path = (data_name, )  # mimicking passing through func(*args)
>>> os.path.join(module_path, *path)
''~/astropy/some/path''
```

So `get_pkg_data_path` is a superset of `_find_pkg_data_path` and doesn't introduce any api changes when passing in a single path argument. The only difference is when multiple path arguments are used.

Reply:
I am just worried that people have been using it not in the advertised way and will run into unexpected behavior with this patch, but I haven't spend too much time thinking about it and unlikely to come back to this in the next few days. Thanks for your patience!

Reply:
I think I might have overthought this one. Seems fine.